ios/RNSSSiriShortcuts.m

```diff
@@ -85,6 +85,18 @@ + (BOOL)application:(UIApplication *)application
     return YES;
 }
 
++ (BOOL)scene:(UIScene *)scene
+continueUserActivity:(NSUserActivity *)userActivity   API_AVAILABLE(ios(13.0)){
+    NSNotificationCenter *notificationCenter = [NSNotificationCenter defaultCenter];
+    [notificationCenter postNotificationName:@"shortcutReceived"
+                                      object:nil
+                                    userInfo:@{
+        @"userInfo": RCTNullIfNil(userActivity.userInfo),
+        @"activityType": userActivity.activityType,
+    }];
```

Comment:
Since this is repeated from the method above, you should extract this portion to it's own function and call it in both of these methods

Reply:
I would (I agree), but I'm afraid this is beyond my expertise in Objective-C.

Reply:
You could do something like:

```objc
+ (void)postShortcutReceivedNotificationWithUserActivity:(NSUserActivity *)userActivity
{
    NSNotificationCenter *notificationCenter = [NSNotificationCenter defaultCenter];
    [notificationCenter postNotificationName:@"shortcutReceived"
                                      object:nil
                                    userInfo:@{
        @"userInfo": RCTNullIfNil(userActivity.userInfo),
        @"activityType": userActivity.activityType,
    }];
}
```

And then in those two methods you can call it like this:

```objc
[RNSSSiriShortcuts postShortcutReceivedNotificationWithUserActivity:userActivity];
```

Reply:
Where would that new function be placed, or is that not important?

Reply:
> Where would that new function be placed, or is that not important?

Could be just before these 2 functions, I guess. Would keep relevant code next to each other